### PR TITLE
Fix osu! logo occasionally being in the wrong state on entering the main menu

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

**Review by smoogipoo — CHANGES_REQUESTED** [review]
You duped code from the state setter and the original code is still there.

**Comment by peppy** [comment]
@smoogipoo nice catch, should be fixed now (along with a few other fixes)

File: osu.Game/Screens/Menu/ButtonSystem.cs [hunk 1]
```diff
@@ -320,9 +282,68 @@ public MenuState State
...
+                        logo.ClearTransforms(targetMember: nameof(Position));
+                        logo.RelativePositionAxes = Axes.Both;
+
+                        logo.MoveTo(new Vector2(0.5f), 800, Easing.OutExpo);
+                        logo.ScaleTo(1, 800, Easing.OutExpo);
+                    }, 150);
+
+                    break;
+                case MenuState.TopLevel:
+                case MenuState.Play:
```
Comment (smoogipoo) [inline osu.Game/Screens/Menu/ButtonSystem.cs]: Are you sure you want to do this in `Play` state? Try pressing spacebar twice in quick succession, looks a bit jarring.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Menu/ButtonSystem.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Menu/ButtonSystem.cs]: mmm, gonna take a bit of thinking to fix that one.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**